### PR TITLE
[doc] Update odop.md

### PR DESCRIPTION
**Pull request by vai9er**
Issue: #

### CODE REVIEW

File: docs/lang/articles/advanced/odop.md [hunk 1]
```diff
@@ -2,10 +2,14 @@
 sidebar_position: 2
 ---
 
-# Objective Data-oriented Programming
+# Objective Data-Oriented Programming
 
 Taichi is a [data-oriented](https://en.wikipedia.org/wiki/Data-oriented_design) programming (DOP) language. However, one-size-fits-all DOP makes modularization hard. To allow modularized code, Taichi borrows some concepts from object-oriented programming (OOP). For convenience, let's call the hybrid scheme objective data-oriented programming (ODOP).
```
[the remark this answers is not in the document]
Comment (vai9er) [reply docs/lang/articles/advanced/odop.md:7]: Fixed!

File: docs/lang/articles/advanced/odop.md [hunk 2]
```diff
@@ -2,10 +2,14 @@
...
 
+:::note
+Data-Oriented Programming approaches coding in a unique way. While you may be familiar with Object-Oriented Programming (OOP), Data-Oriented Design indicates that everything is data that can be acted on. This differentiates functionality from data. They are no longer linked by a set of rules. Your DOP routines are general-purpose and deal with enormous volumes of data. To guarantee that the function takes as little effort as possible, you should organise the data as close to the output data as possible.
```
[the remark this answers is not in the document]
Comment (vai9er) [reply docs/lang/articles/advanced/odop.md:10]: Fixed!

File: docs/lang/articles/advanced/odop.md [hunk 3]
```diff
@@ -2,11 +2,15 @@
 sidebar_position: 2
 ---
 
-# Objective Data-oriented Programming
+# Objective Data-Oriented Programming
 
-Taichi is a [data-oriented](https://en.wikipedia.org/wiki/Data-oriented_design) programming (DOP) language. However, one-size-fits-all DOP makes modularization hard. To allow modularized code, Taichi borrows some concepts from object-oriented programming (OOP). For convenience, let's call the hybrid scheme objective data-oriented programming (ODOP).
+Taichi is a [Data-Oriented](https://en.wikipedia.org/wiki/Data-Oriented_design) programming (DOP) language. However, one-size-fits-all DOP makes modularization hard. To allow modularized code, Taichi borrows some concepts from Object-Oriented programming (OOP). For convenience, let's call the hybrid scheme objective Data-Oriented programming (ODOP).
```
Comment (Olinaaaloompa) [inline docs/lang/articles/advanced/odop.md:7]:
```suggestion
Taichi is a [Data-Oriented](https://en.wikipedia.org/wiki/Data-Oriented_design) Programming (DOP) language. However, one-size-fits-all DOP makes modularization hard. To allow modularized code, Taichi borrows some concepts from Object-Oriented Programming (OOP). For convenience, let's call the hybrid scheme Objective Data-Oriented Programming (ODOP).
```

File: docs/lang/articles/advanced/odop.md [hunk 4]
```diff
@@ -2,11 +2,15 @@
...
+
+:::note
+Data-Oriented Programming approaches coding in a unique way. While you may be familiar with Object-Oriented Programming (OOP), Data-Oriented Design indicates that everything is data that can be acted on. This differentiates functionality from data. They are no longer linked by a set of rules. Your DOP routines are general-purpose and deal with enormous volumes of data. To guarantee that the function takes as little effort as possible, you should organize the data as close to the output data as possible.
```
Comment (Olinaaaloompa) [inline docs/lang/articles/advanced/odop.md:10]:
```suggestion
DOP approaches coding in a unique way. While you may be familiar with OOP, the Data-Oriented design indicates that everything is data that can be acted on. This differentiates functionality from data. They are no longer linked by a set of rules. Your DOP routines are general-purpose and deal with enormous volumes of data. To guarantee that the function takes as little effort as possible, you should organize the data as close to the output data as possible.
```

File: docs/lang/articles/advanced/odop.md [hunk 5]
```diff
@@ -2,11 +2,15 @@
...
 
-- Decorated with `@ti.data_oriented`, a data-oriented class is used when your data is actively updated in the Python scope (such as current time and user input events) and tracked in Taichi kernels. This type of class can have native Python objects as members and must be instantiated in the Python scope. [Data-oriented Class](./data_oriented_class.md) describes this type of class.
+- Decorated with `@ti.data_oriented`, a Data-Oriented class is used when your data is actively updated in the Python scope (such as current time and user input events) and tracked in Taichi kernels. This type of class can have native Python objects as members and must be instantiated in the Python scope. [Data-Oriented Class](./data_oriented_class.md) describes this type of class.
 - Decorated with `@ti.dataclass`, a dataclass is a wrapper of `ti.types.struct`. A dataclass provides more flexibilities. You can define Taichi functions as its methods and call these methods in the Taichi scope. [Data Class](./dataclass.md) describes this type of class.
```
Comment (Olinaaaloompa) [inline docs/lang/articles/advanced/odop.md:16]:
```suggestion
- Decorated with `@ti.dataclass`, a dataclass is a wrapper of `ti.types.struct`. A dataclass provides more flexibilities. You can define Taichi functions as its methods and call these methods in the Taichi scope. [Taichi Dataclass](./dataclass.md) describes this type of class.
```

**Review by Olinaaaloompa — APPROVED** [review]
LGTM!

**Review by neozhaoliang — APPROVED** [review]
LGTM!

**Pull request merged**